Darwin artifacts from mac4n6 project

pstirparo opened the pull request:
I did the run_test and removed couple of duplicates, however it still fails due to the new labels I have introduced for macro categories: Mail, Cloud, Account.

Artifacts from mac4n6 refer only to FILE type and do not have/include any command or other types so far.
Looking forward to your feedback.
Pasquale

<!-- Reviewable:start -->
[<img src="https://reviewable.io/review_button.png" height=40 alt="Review on Reviewable"/>](https://reviewable.io/reviews/forensicartifacts/artifacts/87)
<!-- Reviewable:end -->

destijl commented:
This is great, thanks for your work. Sorry for the slow review I was out of the office for a bit. There's quite a few comments here, but most are pretty easy changes. I'm willing to take on the changes if this seems like too much work, but I wanted to give you the chance to reply to my suggestions first.

@joachimmetz there are a couple of wider questions we need to solve, I mentioned you where you should look.

---

Reviewed 1 of 1 files at r1.
Review status: all files reviewed at latest revision, 31 unresolved discussions, some commit checks failed.

---

<sup>**[definitions/darwin.yaml, line 5 \[r1\]](https://reviewable.io:443/reviews/forensicartifacts/artifacts/87#-Jyj2cjOEPwrMzQV8k7d)** ([raw file](https://github.com/forensicartifacts/artifacts/blob/1926db7ab44ebd979776bc2d41733388e74986b0/definitions/darwin.yaml#L5)):</sup>
Lets remove all of this:

- We'll put you (and a link to your site if you like) in acknowledgements.
- The forensicswiki references should go into each artifact. The general page links aren't particularly useful here. If you add URLs to each artifact then they are displayed in the GRR UI and can take people directly to extra information.
- Last update is handled by git, no need to add that in here, and people will always forget to update it anyway.

---

<sup>**[definitions/darwin.yaml, line 11 \[r1\]](https://reviewable.io:443/reviews/forensicartifacts/artifacts/87#-Jyj3_kEr9gb2penpaLl)** ([raw file](https://github.com/forensicartifacts/artifacts/blob/9b3556b17b00395222271953a86d645286a03a4d/definitions/darwin.yaml#L11)):</sup>
why remove this? It's definitely correct.

---

<sup>**[definitions/darwin.yaml, line 14 \[r1\]](https://reviewable.io:443/reviews/forensicartifacts/artifacts/87#-Jyj3mB08LUTbvQB2PWa)** ([raw file](https://github.com/forensicartifacts/artifacts/blob/9b3556b17b00395222271953a86d645286a03a4d/definitions/darwin.yaml#L14)):</sup>
We should be making these more specific, not removing. This one should be:
http://forensicswiki.org/wiki/Mac_OS_X#Launch_Agents
and that article should point to:

https://developer.apple.com/library/mac/documentation/MacOSX/Conceptual/BPSystemStartup/Chapters/CreatingLaunchdJobs.html
I'll update forensicswiki shortly...

---

<sup>**[definitions/darwin.yaml, line 28 \[r1\]](https://reviewable.io:443/reviews/forensicartifacts/artifacts/87#-Jyj5pUFYdzydc4np4mw)** ([raw file](https://github.com/forensicartifacts/artifacts/blob/9b3556b17b00395222271953a86d645286a03a4d/definitions/darwin.yaml#L28)):</sup>
same as above, the URL should stay.

---

<sup>**[definitions/darwin.yaml, line 30 \[r1\]](https://reviewable.io:443/reviews/forensicartifacts/artifacts/87#-Jyj7vOG_t5sxHSMqoBG)** ([raw file](https://github.com/forensicartifacts/artifacts/blob/1926db7ab44ebd979776bc2d41733388e74986b0/definitions/darwin.yaml#L30)):</sup>
files

---

<sup>**[definitions/darwin.yaml, line 38 \[r1\]](https://reviewable.io:443/reviews/forensicartifacts/artifacts/87#-Jyj61UFjuzIFb7KzpTj)** ([raw file](https://github.com/forensicartifacts/artifacts/blob/1926db7ab44ebd979776bc2d41733388e74986b0/definitions/darwin.yaml#L38)):</sup>
Lets add URLs for all of these that point to the relevant parts of the forensicswiki page. And hope that page gets a fair bit more work :)

---

<sup>**[definitions/darwin.yaml, line 40 \[r1\]](https://reviewable.io:443/reviews/forensicartifacts/artifacts/87#-Jyj8RbEx--bf2f2B9Wc)** ([raw file](https://github.com/forensicartifacts/artifacts/blob/1926db7ab44ebd979776bc2d41733388e74986b0/definitions/darwin.yaml#L40)):</sup>
Lets keep this as OSXSystemLogs

think general doesn't add much, and we should avoid breaking any meta-artifacts people have based on this.

---

<sup>**[definitions/darwin.yaml, line 81 \[r1\]](https://reviewable.io:443/reviews/forensicartifacts/artifacts/87#-Jyj6Q7rHUoOARNxeBfS)** ([raw file](https://github.com/forensicartifacts/artifacts/blob/1926db7ab44ebd979776bc2d41733388e74986b0/definitions/darwin.yaml#L81)):</sup>
This should be recursive to catch the ones that are in subdirs
/Library/Preferences/**

---

<sup>**[definitions/darwin.yaml, line 85 \[r1\]](https://reviewable.io:443/reviews/forensicartifacts/artifacts/87#-Jyj6a82x0mMuJxLG7bH)** ([raw file](https://github.com/forensicartifacts/artifacts/blob/1926db7ab44ebd979776bc2d41733388e74986b0/definitions/darwin.yaml#L85)):</sup>
OSXGlobalPreferences

---

<sup>**[definitions/darwin.yaml, line 89 \[r1\]](https://reviewable.io:443/reviews/forensicartifacts/artifacts/87#-Jyj7CJe1xGzm7Z5c8Uw)** ([raw file](https://github.com/forensicartifacts/artifacts/blob/1926db7ab44ebd979776bc2d41733388e74986b0/definitions/darwin.yaml#L89)):</sup>
In the style guide we put these on a single line, it saves quite a bit of space. Here and all of the others below that are just a single file. "Style note: where sources take a single argument with a single value, the one-line {} form should be used to save on line breaks". https://github.com/ForensicArtifacts/artifacts/blob/master/docs/Artifacts%20definition%20format%20and%20style%20guide.asciidoc

---

<sup>**[definitions/darwin.yaml, line 104 \[r1\]](https://reviewable.io:443/reviews/forensicartifacts/artifacts/87#-Jyj6iKJ5-WLfA2XVnB7)** ([raw file](https://github.com/forensicartifacts/artifacts/blob/1926db7ab44ebd979776bc2d41733388e74986b0/definitions/darwin.yaml#L104)):</sup>
Bluetooth preferences and paired device info

---

<sup>**[definitions/darwin.yaml, line 167 \[r1\]](https://reviewable.io:443/reviews/forensicartifacts/artifacts/87#-Jyj7-CGjbXKQ32wKVv1)** ([raw file](https://github.com/forensicartifacts/artifacts/blob/1926db7ab44ebd979776bc2d41733388e74986b0/definitions/darwin.yaml#L167)):</sup>
Kernel extensions

---

<sup>**[definitions/darwin.yaml, line 243 \[r1\]](https://reviewable.io:443/reviews/forensicartifacts/artifacts/87#-Jyj7lhv2xBSX6S45Y-k)** ([raw file](https://github.com/forensicartifacts/artifacts/blob/1926db7ab44ebd979776bc2d41733388e74986b0/definitions/darwin.yaml#L243)):</sup>
OSXHostsFile

---

<sup>**[definitions/darwin.yaml, line 270 \[r1\]](https://reviewable.io:443/reviews/forensicartifacts/artifacts/87#-JyjADVvZIxdQHoYhC2B)** ([raw file](https://github.com/forensicartifacts/artifacts/blob/1926db7ab44ebd979776bc2d41733388e74986b0/definitions/darwin.yaml#L270)):</sup>
So this one is a problem because we actually have this already in GRR:
https://github.com/google/grr/blob/35da6a71179a4b3e8631620e9bfa1a2e3fc68e0f/artifacts/flow_templates/darwin.yaml#L31

The reason was that the intention here is not really to point at a file to be collected, but rather to tell GRR to list files. So when I first split the artifacts that didn't seem useful outside of GRR, but it probably is. So I think probably what we should do is add DIRECTORY as a type here (or maybe use PATH @joachimmetz):
https://github.com/ForensicArtifacts/artifacts/blob/master/artifacts/definitions.py#L8

 and remove the GRR version. This should be converted to DIRECTORY, and all of the similar ones below (music pictures etc.)

---

<sup>**[definitions/darwin.yaml, line 360 \[r1\]](https://reviewable.io:443/reviews/forensicartifacts/artifacts/87#-Jyj95WqCmcJ91KUjEq8)** ([raw file](https://github.com/forensicartifacts/artifacts/blob/1926db7ab44ebd979776bc2d41733388e74986b0/definitions/darwin.yaml#L360)):</sup>
OSXUserPreferences

---

<sup>**[definitions/darwin.yaml, line 375 \[r1\]](https://reviewable.io:443/reviews/forensicartifacts/artifacts/87#-JyjLjlI5HQaPnYPC8HQ)** ([raw file](https://github.com/forensicartifacts/artifacts/blob/1926db7ab44ebd979776bc2d41733388e74986b0/definitions/darwin.yaml#L375)):</sup>
I think we need to differentiate Account and Authentication. Perhaps "ExternalAccount" would be more accurate. @joachimmetz.

---

<sup>**[definitions/darwin.yaml, line 379 \[r1\]](https://reviewable.io:443/reviews/forensicartifacts/artifacts/87#-JyjED7nCE4QJtkbkcj3)** ([raw file](https://github.com/forensicartifacts/artifacts/blob/1926db7ab44ebd979776bc2d41733388e74986b0/definitions/darwin.yaml#L379)):</sup>
This one definitely needs a link or a longer doc string. Not obvious relationship to external media.

Should also be '%%users.homedir%%/Library/Preferences/com.apple.sidebarlists.plist'

---

<sup>**[definitions/darwin.yaml, line 527 \[r1\]](https://reviewable.io:443/reviews/forensicartifacts/artifacts/87#-JyjFfy-VWjnGloNxV66)** ([raw file](https://github.com/forensicartifacts/artifacts/blob/1926db7ab44ebd979776bc2d41733388e74986b0/definitions/darwin.yaml#L527)):</sup>
This should probably be a DIRECTORY, I assume the intention is just to list files inside?

---

<sup>**[definitions/darwin.yaml, line 560 \[r1\]](https://reviewable.io:443/reviews/forensicartifacts/artifacts/87#-JyjH43ll3ILoL4DuwoL)** ([raw file](https://github.com/forensicartifacts/artifacts/blob/1926db7ab44ebd979776bc2d41733388e74986b0/definitions/darwin.yaml#L560)):</sup>
Is there much value in this one? the specifics below seem more useful. If the intention is "everything in the skype dir" it should be ** to go 3 levels deep. I think we can probably just remove this one.

---

<sup>**[definitions/darwin.yaml, line 605 \[r1\]](https://reviewable.io:443/reviews/forensicartifacts/artifacts/87#-JyjI2R9F3ENtVnnoHS5)** ([raw file](https://github.com/forensicartifacts/artifacts/blob/1926db7ab44ebd979776bc2d41733388e74986b0/definitions/darwin.yaml#L605)):</sup>
Since safari at one point did work on windows we have existing safari artifacts here:
https://github.com/ForensicArtifacts/artifacts/blob/master/definitions/webbrowser.yaml#L304

It think it makes more sense to keep the webbrowsing ones over there, since we've already broken them out. So lets move them into that file.

---

<sup>**[definitions/darwin.yaml, line 606 \[r1\]](https://reviewable.io:443/reviews/forensicartifacts/artifacts/87#-JyjIpJ3jV8wV6rLCfOF)** ([raw file](https://github.com/forensicartifacts/artifacts/blob/1926db7ab44ebd979776bc2d41733388e74986b0/definitions/darwin.yaml#L606)):</sup>
These generic directory ones seem very low value, and are probably just clutter. I think we can just remove them. If there's something specific that is interesting under here we should add another artifact.

---

<sup>**[definitions/darwin.yaml, line 805 \[r1\]](https://reviewable.io:443/reviews/forensicartifacts/artifacts/87#-JyjJrxltLzZLT65klr4)** ([raw file](https://github.com/forensicartifacts/artifacts/blob/1926db7ab44ebd979776bc2d41733388e74986b0/definitions/darwin.yaml#L805)):</sup>
Same deal for firefox.

---

<sup>**[definitions/darwin.yaml, line 917 \[r1\]](https://reviewable.io:443/reviews/forensicartifacts/artifacts/87#-JyjK356vrDsR_PwT7hp)** ([raw file](https://github.com/forensicartifacts/artifacts/blob/1926db7ab44ebd979776bc2d41733388e74986b0/definitions/darwin.yaml#L917)):</sup>
remove generic directories and move to webbrowser file (sync with what's there).

---

<sup>**[definitions/darwin.yaml, line 1052 \[r1\]](https://reviewable.io:443/reviews/forensicartifacts/artifacts/87#-JyjKMMSTmIfBTKu4KyZ)** ([raw file](https://github.com/forensicartifacts/artifacts/blob/1926db7ab44ebd979776bc2d41733388e74986b0/definitions/darwin.yaml#L1052)):</sup>
I don't use the mail app, but I suspect this should be:
'%%users.homedir%%/Library/Mail/V2/IMAP-**'

since name@address is an actual address.

---

<sup>**[definitions/darwin.yaml, line 1061 \[r1\]](https://reviewable.io:443/reviews/forensicartifacts/artifacts/87#-JyjKc1p_aRPrMzXhDwu)** ([raw file](https://github.com/forensicartifacts/artifacts/blob/1926db7ab44ebd979776bc2d41733388e74986b0/definitions/darwin.yaml#L1061)):</sup>
same here.

---

<sup>**[definitions/darwin.yaml, line 1101 \[r1\]](https://reviewable.io:443/reviews/forensicartifacts/artifacts/87#-JyjKhhWnUHXPHUG1BMe)** ([raw file](https://github.com/forensicartifacts/artifacts/blob/1926db7ab44ebd979776bc2d41733388e74986b0/definitions/darwin.yaml#L1101)):</sup>
OSXMailDownloadedAttachments

---

<sup>**[definitions/darwin.yaml, line 1137 \[r1\]](https://reviewable.io:443/reviews/forensicartifacts/artifacts/87#-JyjKuDeAZ5THrP4cljq)** ([raw file](https://github.com/forensicartifacts/artifacts/blob/1926db7ab44ebd979776bc2d41733388e74986b0/definitions/darwin.yaml#L1137)):</sup>
remove these, they will get stale.

grep "name" | wc -l

works just as well

---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/forensicartifacts/artifacts/87)
<!-- Sent from Reviewable.io -->

destijl commented:
<sup>**[definitions/darwin.yaml, line 375 \[r1\]](https://reviewable.io:443/reviews/forensicartifacts/artifacts/87#-JyjLjlI5HQaPnYPC8HQ)** ([raw file](https://github.com/forensicartifacts/artifacts/blob/1926db7ab44ebd979776bc2d41733388e74986b0/definitions/darwin.yaml#L375)):</sup>
Oh, and can you add the new labels here:
https://github.com/ForensicArtifacts/artifacts/blob/master/artifacts/definitions.py#L14

so the tests pass?

---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/forensicartifacts/artifacts/87)
<!-- Sent from Reviewable.io -->

pstirparo commented:
Hi Greg,
Sorry for the delay, I was on holiday till last week. I will start reviewing the comments Reviewable on monday, but from a quick read they sound all reasonable to me.

joachimmetz commented:
Review status: all files reviewed at latest revision, 26 unresolved discussions, some commit checks failed.

---

<sup>**[definitions/darwin.yaml, line 270 \[r1\]](https://reviewable.io:443/reviews/forensicartifacts/artifacts/87#-JyjADVvZIxdQHoYhC2B)** ([raw file](https://github.com/forensicartifacts/artifacts/blob/1926db7ab44ebd979776bc2d41733388e74986b0/definitions/darwin.yaml#L270)):</sup>
The DIRECTORY the solution SGTM, I'd had introduced PATH for that purposes.

---

<sup>**[definitions/darwin.yaml, line 375 \[r1\]](https://reviewable.io:443/reviews/forensicartifacts/artifacts/87#-JyjLjlI5HQaPnYPC8HQ)** ([raw file](https://github.com/forensicartifacts/artifacts/blob/1926db7ab44ebd979776bc2d41733388e74986b0/definitions/darwin.yaml#L375)):</sup>
So this artifact contains more than just account information. I'm fine with Account if we get add a clear definition of its purpose: https://github.com/ForensicArtifacts/artifacts/blob/master/docs/Artifacts%20definition%20format%20and%20style%20guide.asciidoc#labels

---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/forensicartifacts/artifacts/87)
<!-- Sent from Reviewable.io -->

pstirparo commented:
Review status: all files reviewed at latest revision, 26 unresolved discussions, some commit checks failed.

---

<sup>**[definitions/darwin.yaml, line 11 \[r1\]](https://reviewable.io:443/reviews/forensicartifacts/artifacts/87#-Jyj3_kEr9gb2penpaLl)** ([raw file](https://github.com/forensicartifacts/artifacts/blob/9b3556b17b00395222271953a86d645286a03a4d/definitions/darwin.yaml#L11)):</sup>
sorry, my bad, I had missed it somehow. fixed

---

<sup>**[definitions/darwin.yaml, line 14 \[r1\]](https://reviewable.io:443/reviews/forensicartifacts/artifacts/87#-Jyj3mB08LUTbvQB2PWa)** ([raw file](https://github.com/forensicartifacts/artifacts/blob/9b3556b17b00395222271953a86d645286a03a4d/definitions/darwin.yaml#L14)):</sup>
I agree. For the time being I just put the URLs back to each artifacts (as for your first general comment), then once the wiki pages will be updated we can put the specific URL for each artifact.

---

<sup>**[definitions/darwin.yaml, line 28 \[r1\]](https://reviewable.io:443/reviews/forensicartifacts/artifacts/87#-Jyj5pUFYdzydc4np4mw)** ([raw file](https://github.com/forensicartifacts/artifacts/blob/9b3556b17b00395222271953a86d645286a03a4d/definitions/darwin.yaml#L28)):</sup>
Done.

---

<sup>**[definitions/darwin.yaml, line 30 \[r1\]](https://reviewable.io:443/reviews/forensicartifacts/artifacts/87#-Jyj7vOG_t5sxHSMqoBG)** ([raw file](https://github.com/forensicartifacts/artifacts/blob/1926db7ab44ebd979776bc2d41733388e74986b0/definitions/darwin.yaml#L30)):</sup>
Done.

---

<sup>**[definitions/darwin.yaml, line 38 \[r1\]](https://reviewable.io:443/reviews/forensicartifacts/artifacts/87#-Jyj61UFjuzIFb7KzpTj)** ([raw file](https://github.com/forensicartifacts/artifacts/blob/1926db7ab44ebd979776bc2d41733388e74986b0/definitions/darwin.yaml#L38)):</sup>
Done.

---

<sup>**[definitions/darwin.yaml, line 40 \[r1\]](https://reviewable.io:443/reviews/forensicartifacts/artifacts/87#-Jyj8RbEx--bf2f2B9Wc)** ([raw file](https://github.com/forensicartifacts/artifacts/blob/1926db7ab44ebd979776bc2d41733388e74986b0/definitions/darwin.yaml#L40)):</sup>
Done.

---

<sup>**[definitions/darwin.yaml, line 81 \[r1\]](https://reviewable.io:443/reviews/forensicartifacts/artifacts/87#-Jyj6Q7rHUoOARNxeBfS)** ([raw file](https://github.com/forensicartifacts/artifacts/blob/1926db7ab44ebd979776bc2d41733388e74986b0/definitions/darwin.yaml#L81)):</sup>
Done.

---

<sup>**[definitions/darwin.yaml, line 85 \[r1\]](https://reviewable.io:443/reviews/forensicartifacts/artifacts/87#-Jyj6a82x0mMuJxLG7bH)** ([raw file](https://github.com/forensicartifacts/artifacts/blob/1926db7ab44ebd979776bc2d41733388e74986b0/definitions/darwin.yaml#L85)):</sup>
Done.

---

<sup>**[definitions/darwin.yaml, line 89 \[r1\]](https://reviewable.io:443/reviews/forensicartifacts/artifacts/87#-Jyj7CJe1xGzm7Z5c8Uw)** ([raw file](https://github.com/forensicartifacts/artifacts/blob/1926db7ab44ebd979776bc2d41733388e74986b0/definitions/darwin.yaml#L89)):</sup>
Done.

---

<sup>**[definitions/darwin.yaml, line 104 \[r1\]](https://reviewable.io:443/reviews/forensicartifacts/artifacts/87#-Jyj6iKJ5-WLfA2XVnB7)** ([raw file](https://github.com/forensicartifacts/artifacts/blob/1926db7ab44ebd979776bc2d41733388e74986b0/definitions/darwin.yaml#L104)):</sup>
Done.

---

<sup>**[definitions/darwin.yaml, line 167 \[r1\]](https://reviewable.io:443/reviews/forensicartifacts/artifacts/87#-Jyj7-CGjbXKQ32wKVv1)** ([raw file](https://github.com/forensicartifacts/artifacts/blob/1926db7ab44ebd979776bc2d41733388e74986b0/definitions/darwin.yaml#L167)):</sup>
Done.

---

<sup>**[definitions/darwin.yaml, line 243 \[r1\]](https://reviewable.io:443/reviews/forensicartifacts/artifacts/87#-Jyj7lhv2xBSX6S45Y-k)** ([raw file](https://github.com/forensicartifacts/artifacts/blob/1926db7ab44ebd979776bc2d41733388e74986b0/definitions/darwin.yaml#L243)):</sup>
Done.

---

<sup>**[definitions/darwin.yaml, line 270 \[r1\]](https://reviewable.io:443/reviews/forensicartifacts/artifacts/87#-JyjADVvZIxdQHoYhC2B)** ([raw file](https://github.com/forensicartifacts/artifacts/blob/1926db7ab44ebd979776bc2d41733388e74986b0/definitions/darwin.yaml#L270)):</sup>
DIRECTORY SGTM too. Done.

---

<sup>**[definitions/darwin.yaml, line 360 \[r1\]](https://reviewable.io:443/reviews/forensicartifacts/artifacts/87#-Jyj95WqCmcJ91KUjEq8)** ([raw file](https://github.com/forensicartifacts/artifacts/blob/1926db7ab44ebd979776bc2d41733388e74986b0/definitions/darwin.yaml#L360)):</sup>
Done.

---

<sup>**[definitions/darwin.yaml, line 375 \[r1\]](https://reviewable.io:443/reviews/forensicartifacts/artifacts/87#-JyjLjlI5HQaPnYPC8HQ)** ([raw file](https://github.com/forensicartifacts/artifacts/blob/1926db7ab44ebd979776bc2d41733388e74986b0/definitions/darwin.yaml#L375)):</sup>
SGTM too.
- replaced "Account" with "ExternalAccount"
- added "ExternalAccount" in the definitions.py file

@joachimmetz I tried to give a definition in the definitions.py file too, let me know what do you think. 

---

<sup>**[definitions/darwin.yaml, line 379 \[r1\]](https://reviewable.io:443/reviews/forensicartifacts/artifacts/87#-JyjED7nCE4QJtkbkcj3)** ([raw file](https://github.com/forensicartifacts/artifacts/blob/1926db7ab44ebd979776bc2d41733388e74986b0/definitions/darwin.yaml#L379)):</sup>
Path corrected. Also, since I add the specific URL as from on of your previous comments, in that URL is explained a bit more: http://forensicswiki.org/wiki/Mac_OS_X_10.9_-_Artifacts_Location#Preferences
Is it enough like this?

---

<sup>**[definitions/darwin.yaml, line 527 \[r1\]](https://reviewable.io:443/reviews/forensicartifacts/artifacts/87#-JyjFfy-VWjnGloNxV66)** ([raw file](https://github.com/forensicartifacts/artifacts/blob/1926db7ab44ebd979776bc2d41733388e74986b0/definitions/darwin.yaml#L527)):</sup>
Done.

---

<sup>**[definitions/darwin.yaml, line 560 \[r1\]](https://reviewable.io:443/reviews/forensicartifacts/artifacts/87#-JyjH43ll3ILoL4DuwoL)** ([raw file](https://github.com/forensicartifacts/artifacts/blob/1926db7ab44ebd979776bc2d41733388e74986b0/definitions/darwin.yaml#L560)):</sup>
removed. Done.

---

<sup>**[definitions/darwin.yaml, line 605 \[r1\]](https://reviewable.io:443/reviews/forensicartifacts/artifacts/87#-JyjI2R9F3ENtVnnoHS5)** ([raw file](https://github.com/forensicartifacts/artifacts/blob/1926db7ab44ebd979776bc2d41733388e74986b0/definitions/darwin.yaml#L605)):</sup>
Browser atrifacts removed, I will sync with webbrowser.yaml later on.
Done.

---

<sup>**[definitions/darwin.yaml, line 606 \[r1\]](https://reviewable.io:443/reviews/forensicartifacts/artifacts/87#-JyjIpJ3jV8wV6rLCfOF)** ([raw file](https://github.com/forensicartifacts/artifacts/blob/1926db7ab44ebd979776bc2d41733388e74986b0/definitions/darwin.yaml#L606)):</sup>
Browser atrifacts removed, I will sync with webbrowser.yaml later on.
Done.

---

<sup>**[definitions/darwin.yaml, line 805 \[r1\]](https://reviewable.io:443/reviews/forensicartifacts/artifacts/87#-JyjJrxltLzZLT65klr4)** ([raw file](https://github.com/forensicartifacts/artifacts/blob/1926db7ab44ebd979776bc2d41733388e74986b0/definitions/darwin.yaml#L805)):</sup>
Browser atrifacts removed, I will sync with webbrowser.yaml later on.
Done.

---

<sup>**[definitions/darwin.yaml, line 917 \[r1\]](https://reviewable.io:443/reviews/forensicartifacts/artifacts/87#-JyjK356vrDsR_PwT7hp)** ([raw file](https://github.com/forensicartifacts/artifacts/blob/1926db7ab44ebd979776bc2d41733388e74986b0/definitions/darwin.yaml#L917)):</sup>
Browser atrifacts removed, I will sync with webbrowser.yaml later on.
Done.

---

<sup>**[definitions/darwin.yaml, line 1052 \[r1\]](https://reviewable.io:443/reviews/forensicartifacts/artifacts/87#-JyjKMMSTmIfBTKu4KyZ)** ([raw file](https://github.com/forensicartifacts/artifacts/blob/1926db7ab44ebd979776bc2d41733388e74986b0/definitions/darwin.yaml#L1052)):</sup>
ok, but shouldn't then be '%%users.homedir%%/Library/Mail/V2/IMAP-*/*' ?

---

<sup>**[definitions/darwin.yaml, line 1061 \[r1\]](https://reviewable.io:443/reviews/forensicartifacts/artifacts/87#-JyjKc1p_aRPrMzXhDwu)** ([raw file](https://github.com/forensicartifacts/artifacts/blob/1926db7ab44ebd979776bc2d41733388e74986b0/definitions/darwin.yaml#L1061)):</sup>
same as above

---

<sup>**[definitions/darwin.yaml, line 1101 \[r1\]](https://reviewable.io:443/reviews/forensicartifacts/artifacts/87#-JyjKhhWnUHXPHUG1BMe)** ([raw file](https://github.com/forensicartifacts/artifacts/blob/1926db7ab44ebd979776bc2d41733388e74986b0/definitions/darwin.yaml#L1101)):</sup>
Done.

---

<sup>**[definitions/darwin.yaml, line 1137 \[r1\]](https://reviewable.io:443/reviews/forensicartifacts/artifacts/87#-JyjKuDeAZ5THrP4cljq)** ([raw file](https://github.com/forensicartifacts/artifacts/blob/1926db7ab44ebd979776bc2d41733388e74986b0/definitions/darwin.yaml#L1137)):</sup>
Done.

---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/forensicartifacts/artifacts/87)
<!-- Sent from Reviewable.io -->

destijl commented:
This looks great. I have a couple of minor comments as below but this is great work, thanks. I also have a small change to make to GRR before we submit to add support for DIRECTORY but it is very close.
<!-- Reviewable comment -K0XKtw0P6nmZmcw3xcm:-370112748 -->

---

Review status: 0 of 2 files reviewed at latest revision, 26 unresolved discussions, all commit checks successful.

---

<sup>**[definitions/darwin.yaml, line 85 \[r1\]](https://reviewable.io:443/reviews/forensicartifacts/artifacts/87#-Jyj6a82x0mMuJxLG7bH)** ([raw file](https://github.com/forensicartifacts/artifacts/blob/1926db7ab44ebd979776bc2d41733388e74986b0/definitions/darwin.yaml#L85)):</sup>
Looks like you missed this one ;)
<!-- Reviewable comment -K0XEz2X_CCuCeTSig4d:13612126 -->

---

<sup>**[definitions/darwin.yaml, line 104 \[r1\]](https://reviewable.io:443/reviews/forensicartifacts/artifacts/87#-Jyj6iKJ5-WLfA2XVnB7)** ([raw file](https://github.com/forensicartifacts/artifacts/blob/1926db7ab44ebd979776bc2d41733388e74986b0/definitions/darwin.yaml#L104)):</sup>
paired typo
<!-- Reviewable comment -K0XF6ei-1egPFD4MO20:179705830 -->

---

<sup>**[definitions/darwin.yaml, line 270 \[r1\]](https://reviewable.io:443/reviews/forensicartifacts/artifacts/87#-JyjADVvZIxdQHoYhC2B)** ([raw file](https://github.com/forensicartifacts/artifacts/blob/1926db7ab44ebd979776bc2d41733388e74986b0/definitions/darwin.yaml#L270)):</sup>
Lets use DIRECTORY instead of PATH. I want to remove PATH, it's way too confusing. I'd like to quickly add DIRECTORY support into GRR before we submit this too. That way we avoid breaking people who import the latest repo into GRR. Apologies for  the delay I'll try and get it done very soon and comment back here when it is in.
<!-- Reviewable comment -K0XFQZFTB375YlTjXa7:-281013211 -->

---

<sup>**[definitions/darwin.yaml, line 379 \[r1\]](https://reviewable.io:443/reviews/forensicartifacts/artifacts/87#-JyjED7nCE4QJtkbkcj3)** ([raw file](https://github.com/forensicartifacts/artifacts/blob/1926db7ab44ebd979776bc2d41733388e74986b0/definitions/darwin.yaml#L379)):</sup>
Better if you don't have to follow the link. Add:
```
This plist contains the names of volumes mounted on the desktop that have appeared in the sidebar list.
```
<!-- Reviewable comment -K0XI2Np4LuRXgRHkBY_:-1220676690 -->

---

<sup>**[definitions/darwin.yaml, line 1052 \[r1\]](https://reviewable.io:443/reviews/forensicartifacts/artifacts/87#-JyjKMMSTmIfBTKu4KyZ)** ([raw file](https://github.com/forensicartifacts/artifacts/blob/1926db7ab44ebd979776bc2d41733388e74986b0/definitions/darwin.yaml#L1052)):</sup>
My syntax and this one are both fine, the only difference is mine will descend 3 levels in the GRR implementation. If you only need the first level this is fine.
<!-- Reviewable comment -K0XKCC-k90e-acx0h9M:1158117601 -->

---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/forensicartifacts/artifacts/87)
<!-- Sent from Reviewable.io -->

destijl commented:
Reviewed 2 of 2 files at r2.
Review status: all files reviewed at latest revision, 26 unresolved discussions, some commit checks failed.

---

<sup>**[definitions/darwin.yaml, line 270 \[r1\]](https://reviewable.io:443/reviews/forensicartifacts/artifacts/87#-JyjADVvZIxdQHoYhC2B)** ([raw file](https://github.com/forensicartifacts/artifacts/blob/1926db7ab44ebd979776bc2d41733388e74986b0/definitions/darwin.yaml#L270)):</sup>
I've added this support into GRR and will export it to opensource in the next day or so. @pstirparo if you can get these PATH ones changed into DIRECTORY artifacts and address the other minor comments this can go in.
<!-- Reviewable comment -K15P5z28d47UuOu4IHV:1901798925 -->

---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/forensicartifacts/artifacts/87)
<!-- Sent from Reviewable.io -->

pstirparo commented:
Review status: all files reviewed at latest revision, 26 unresolved discussions, some commit checks failed.

---

<sup>**[definitions/darwin.yaml, line 85 \[r1\]](https://reviewable.io:443/reviews/forensicartifacts/artifacts/87#-Jyj6a82x0mMuJxLG7bH)** ([raw file](https://github.com/forensicartifacts/artifacts/blob/1926db7ab44ebd979776bc2d41733388e74986b0/definitions/darwin.yaml#L85)):</sup>
Ups, sorry, fixed. I realised that there was another OSXGlobalPreferences in the users artifacts, so I changed that to OSXUserGlobalPreferences to avoid confusion
<!-- Reviewable comment -K16AuGbimBqC6PXWG9L:-981113220 -->

---

<sup>**[definitions/darwin.yaml, line 104 \[r1\]](https://reviewable.io:443/reviews/forensicartifacts/artifacts/87#-Jyj6iKJ5-WLfA2XVnB7)** ([raw file](https://github.com/forensicartifacts/artifacts/blob/1926db7ab44ebd979776bc2d41733388e74986b0/definitions/darwin.yaml#L104)):</sup>
Done.
<!-- Reviewable comment -K169wQKczyjcprHBhvQ:-499146009 -->

---

<sup>**[definitions/darwin.yaml, line 270 \[r1\]](https://reviewable.io:443/reviews/forensicartifacts/artifacts/87#-JyjADVvZIxdQHoYhC2B)** ([raw file](https://github.com/forensicartifacts/artifacts/blob/1926db7ab44ebd979776bc2d41733388e74986b0/definitions/darwin.yaml#L270)):</sup>
Done. Sorry for the late reply I've been travelling a bit lately.
<!-- Reviewable comment -K16BpQ3xGQdf9A0jmkO:-2054856184 -->

---

<sup>**[definitions/darwin.yaml, line 379 \[r1\]](https://reviewable.io:443/reviews/forensicartifacts/artifacts/87#-JyjED7nCE4QJtkbkcj3)** ([raw file](https://github.com/forensicartifacts/artifacts/blob/1926db7ab44ebd979776bc2d41733388e74986b0/definitions/darwin.yaml#L379)):</sup>
Done. I added in the "doc", I followed the style from the others yaml files
<!-- Reviewable comment -K16D75A1D--WnEBOmRJ:551860679 -->

---

<sup>**[definitions/darwin.yaml, line 1052 \[r1\]](https://reviewable.io:443/reviews/forensicartifacts/artifacts/87#-JyjKMMSTmIfBTKu4KyZ)** ([raw file](https://github.com/forensicartifacts/artifacts/blob/1926db7ab44ebd979776bc2d41733388e74986b0/definitions/darwin.yaml#L1052)):</sup>
Yes, it's enough one level because there there are the .mbox files to grab... then all the other files are inside the mbox 
<!-- Reviewable comment -K16E9X14VXVuGQSr9_l:-126685089 -->

---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/forensicartifacts/artifacts/87)
<!-- Sent from Reviewable.io -->

destijl commented:
Awesome, this can go in once you have merged with master, looks like there are some conflicts that need to be resolved.
<!-- Reviewable comment -K16GxOiMSqz7-mL7sm1:1406837616 -->

---

Review status: 1 of 2 files reviewed at latest revision, 2 unresolved discussions, some commit checks failed.

---



Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/forensicartifacts/artifacts/87)
<!-- Sent from Reviewable.io -->

pstirparo commented:
The only error I get running run_tests.py is due to the missing DIRECTORY support. I tried to add in definition.py but still does not sees is (while for the new labels it worked). Am I missing something else?
<!-- Reviewable comment -K16I17lvLe_AXIvG0Da:-1203183198 -->

---

Review status: 1 of 2 files reviewed at latest revision, 2 unresolved discussions, some commit checks failed.

---



Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/forensicartifacts/artifacts/87)
<!-- Sent from Reviewable.io -->

destijl commented:
Yeah, you actually need to add the validation code. Here:
https://github.com/ForensicArtifacts/artifacts/blob/master/artifacts/artifact.py#L60

and here:

https://github.com/ForensicArtifacts/artifacts/blob/master/artifacts/artifact.py#L60

I can do this for you if you like.
<!-- Reviewable comment -K16JdLmyy5hBvxjeL63:495647355 -->

---

Review status: 1 of 2 files reviewed at latest revision, 2 unresolved discussions, some commit checks failed.

---



Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/forensicartifacts/artifacts/87)
<!-- Sent from Reviewable.io -->

destijl commented:
Copy paste fail, here:
https://github.com/ForensicArtifacts/artifacts/blob/master/artifacts/source_type.py#L90
<!-- Reviewable comment -K16KeK49G6aoiyklcZW:1444587230 -->

---

Review status: 1 of 2 files reviewed at latest revision, 2 unresolved discussions, some commit checks failed.

---



Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/forensicartifacts/artifacts/87)
<!-- Sent from Reviewable.io -->

pstirparo commented:
Done, now it passes all the tests. Let me know if it' s ok or I can do something else. Thanks for the support
<!-- Reviewable comment -K16MB0CaTMxt9rAIrsm:-1820787062 -->

---

Review status: 1 of 4 files reviewed at latest revision, 2 unresolved discussions, some commit checks failed.

---



Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/forensicartifacts/artifacts/87)
<!-- Sent from Reviewable.io -->

destijl commented:
cool, just needs a merge. Follow the "use the command line" instructions from github above.

destijl commented:
The GRR change for DIRECTORY is now in open source. You're so close, this will probably take 2 minutes to fix any conflicts. I'd do it, but you should get the github credit :)

pstirparo commented:
I don't see the conflicts on my command line and looks like not able to merge:
Argo:artifacts paco$ git status
On branch master
Your branch is up-to-date with 'origin/master'.
nothing to commit, working directory clean

Argo:artifacts paco$ git merge origin/master master
Already up-to-date.

Argo:artifacts paco$ git remote -v
origin	https://github.com/pstirparo/artifacts.git (fetch)
origin	https://github.com/pstirparo/artifacts.git (push)
upstream	https://github.com/ForensicArtifacts/artifacts.git (fetch)
upstream	https://github.com/ForensicArtifacts/artifacts.git (push)

Argo:artifacts paco$ git remote show origin
* remote origin
  Fetch URL: https://github.com/pstirparo/artifacts.git
  Push  URL: https://github.com/pstirparo/artifacts.git
  HEAD branch: master
  Remote branch:
    master tracked
  Local branch configured for 'git pull':
    master merges with remote master
  Local ref configured for 'git push':
    master pushes to master (up to date)

any hint? tnx
<!-- Reviewable comment -K1GtMA18zBRVkJJuxQT:756871054 -->

---

Review status: 1 of 4 files reviewed at latest revision, 2 unresolved discussions, some commit checks failed.

---



Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/forensicartifacts/artifacts/87)
<!-- Sent from Reviewable.io -->

destijl commented:
I usually do `git fetch upstream && git checkout master && git merge upstream/master`

joachimmetz commented:
or you can use:
```
git fetch upstream && git pull --rebase upstream master
```

pstirparo commented:
I had some issue when using rebase
```
git fetch upstream && git pull --rebase upstream master
```
I got conflicts message and the yaml file went back to an older version (?!?!), so I aborted it (I probably messed up something). I re-cloned my repo, tried to fix manually with the new yaml on main repo and then tried again with Greg's suggestion, now seems working and no more conflicts.
```
Argo:artifacts paco$ git fetch upstream
Argo:artifacts paco$ git checkout master
Already on 'master'
Your branch is up-to-date with 'origin/master'.
Argo:artifacts paco$ git merge upstream/master
Already up-to-date.
```
Hopefully I didn't messed it up too much, let me know if it's ok.
tnx

destijl commented:
Looks good \o/
